windows/RNDocumentPicker/RCTDocumentPickerModule.cs

```diff
@@ -0,0 +1,188 @@
+                if (fileTypeArray != null && fileTypeArray.Count > 0)
+                {
+                    foreach (String typeString in fileTypeArray)
+                    {
+                        List<String> types = typeString.Split(' ').ToList();
+                        foreach (String type in types)
+                        {
+                            if (Regex.Match(type, "(^[.]+[A-Za-z0-9]*$)|(^[*]$)").Success)
+                            {
+                                openPicker.FileTypeFilter.Add(type);
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    openPicker.FileTypeFilter.Add("*");
+                }
+
+                RunOnDispatcher(async () =>
+                {
+                    try
+                    {
+                        if (_isInForeground)
+                        {
+                            var isMultiple = options.Value<bool>(OPTION_MULIPLE);
+                            if (isMultiple)
+                            {
+                                await PickMultipleFileAsync(openPicker, promise);
+                            }
+                            else
+                            {
+                                await PickSingleFileAsync(openPicker, promise);
+                            }
+                        }
+                        else
+                        {
+                            _pendingPicker = openPicker;
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        promise.Reject(E_FAILED_TO_SHOW_PICKER, E_FAILED_TO_SHOW_PICKER);
+                    }
+                });
+            }
+            catch (Exception) {
+                promise.Reject(E_UNEXPECTED_EXCEPTION, E_UNEXPECTED_EXCEPTION);
+            }
+        }
+
+        private async Task<JObject> PrepareFile(StorageFile file)
+        {
+            var basicProperties = await file.GetBasicPropertiesAsync();
+
+            return new JObject {
+                    { FIELD_URI, file.Path },
+                    { FIELD_TYPE, file.ContentType },
+                    { FIELD_NAME, file.Name },
+                    { FIELD_SIZE, basicProperties.Size}
+                };
+        }
+
+        private async Task<bool> PickMultipleFileAsync(FileOpenPicker picker, IPromise promise) {
+            IReadOnlyList<StorageFile> files = await picker.PickMultipleFilesAsync().AsTask().ConfigureAwait(false);
+            if (files.Count > 0)
+            {
+                JArray jarrayObj = new JArray();
+                foreach (var file in files) {
+                    jarrayObj.Add(PrepareFile(file).Result);
+                }
+                promise.Resolve(jarrayObj);
+            }
+            else
+            {
+                promise.Reject(E_DOCUMENT_PICKER_CANCELED, E_DOCUMENT_PICKER_CANCELED);
```

Comment:
Use the same `"User canceled document picker"` message as I did in Android instead of duplicating the key.